llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -223,6 +223,12 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
     setCondCodeAction(ISD::SETUNE, VT, Expand);
   }
 
+  if (Subtarget.hasAVX10_2()) {
+    for (auto VT : {MVT::f16, MVT::f32, MVT::f64}) {
+      setCondCodeAction(ISD::SETOEQ, VT, Custom);
+      setCondCodeAction(ISD::SETUNE, VT, Custom);
+    }
+  }
```

Comment:
Move this under FP16 code. We need to override `f16` with this code.

Reply:
you mean only FP16 or Complete SetCondCodeAction for f16, f32 and f64?

Reply:
All of them. See the line number of previous diff.

Reply:
https://github.com/llvm/llvm-project/pull/113905 I tried this. Moved SETOEQ and SETUNE group after avx10_2 & softfloat feature. it did not work.